### PR TITLE
GH-53 Create PeerManager and add new Peer

### LINKED ISSUE #53: Create PeerManager and add new Peer
- [x] Build PeerManager GenServer
- [x] `add_peer` function to expect a ip address
- [x] on adding peer `/ping` to see if it is alive, if so, add to a list of peers
- [x] `check_peers` function to check all peers in list if they are still alive, remove if not

### PR DESCRIPTION
**Pull request by d-velev**
resolves #53 , #52 , #41

### CODE REVIEW

File: apps/aecore/lib/aecore/peers/worker.ex [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      {:reply, :ok, List.delete(peers, uri)}
+    else
+      {:reply, :error, peers}
+    end
+  end
+
+  def handle_call(:check_peers, _from, peers) do
+    updated_peers = Enum.filter(peers, fn(peer) ->
+      reply = Client.ping_uri(peer)
+      :ok = reply
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/peers/worker.ex]: shouldn't this be `==`

File: apps/aecore/lib/aecore/peers/worker.ex [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  @spec all_peers() :: list()
+  def all_peers() do
+    GenServer.call(__MODULE__, :all_peers)
+  end
+
+  def handle_call({:add_peer,uri}, _from, peers) do
+    case(Client.ping_uri(uri)) do
+      :ok ->
+        {:reply, :ok, [uri | peers]}
```
Comment (thepiwo) [inline apps/aecore/lib/aecore/peers/worker.ex]: can you avoid adding the same peer, twice

File: apps/aehttpserver/config/config.exs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# and its dependencies with the aid of the Mix.Config module.
+#
+# This configuration file is loaded before any dependency and
+# is restricted to this project.
+use Mix.Config
+
+# Configures the endpoint
+config :aehttpserver, Aehttpserver.Endpoint,
+  url: [host: "localhost"],
+  secret_key_base: "5OCr9Ljxr0s8cSx36fmjKgi6hWETJeoxH+jMvBSNDCO/pt0Zqo1S0+EGGJ2VyOoa",
```
Comment (thepiwo) [inline apps/aehttpserver/config/config.exs]: Is this auto-generated? If we don't need it, please remove it

File: apps/aehttpserver/web/controllers/page_controller.ex [hunk 4]
```diff
@@ -0,0 +1,7 @@
+defmodule Aehttpserver.PageController do
```
Comment (thepiwo) [inline apps/aehttpserver/web/controllers/page_controller.ex]: we don't need this

**Comment by thepiwo** [comment]
please remove all code used for templating, as I see we don't need `gettext` at all.
`apps/aehttpserver/priv/` and `apps/aehttpserver/web/` should mostly be unnecessary.

**Comment by d-velev** [comment]
don't merge, requested changes will be applied in #69

**Pull request closed without merging**